### PR TITLE
ci: make self-hosted actions-cache steps access S3 from correct region

### PR DESCRIPTION
**Pull request by strophy**
## Issue being fixed or feature implemented

#1398 reverted #1391 without resolving the original issue

## What was done?

Add a condition the AWS login step so it only occurs on AWS self-hosted runner infra

## How Has This Been Tested?

In CI

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: .github/actions/nodejs/action.yaml [hunk 1]
```diff
@@ -17,6 +17,11 @@ runs:
       shell: bash
       run: npm config set audit false
 
+    - uses: aws-actions/configure-aws-credentials@v4
+      if: contains(runner.name, 'ubuntu-platform')
+      with:
+        aws-region: eu-west-1
+
     - name: Cache NPM build artifacts
```
Comment (shumkov) [inline .github/actions/nodejs/action.yaml:25]: We still keep it? Shall we use action/cache if aren't in AWS?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: .github/actions/nodejs/action.yaml (hunk 1 above)
Comment (strophy) [reply .github/actions/nodejs/action.yaml:26]: Good point, added a condition here as well

File: .github/actions/rust/action.yaml [hunk 2]
```diff
@@ -32,8 +32,15 @@ runs:
         echo "PROTOC=${HOME}/.local/bin/protoc" >> $GITHUB_ENV
         export PATH="${PATH}:${HOME}/.local/bin"
 
+    - name: Configure AWS credentials and bucket region
+      uses: aws-actions/configure-aws-credentials@v4
+      if: contains(runner.name, 'ubuntu-platform')
+      with:
+        aws-region: eu-west-1
+
     - name: Cache Rust build artifacts
```
Comment (shumkov) [inline .github/actions/rust/action.yaml:41]: Let's add action/cache otherwise (`!contains(runner.name, 'ubuntu-platform')`)?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: .github/actions/rust/action.yaml (hunk 2 above)
Comment (strophy) [reply .github/actions/rust/action.yaml]: Ah now I understand what you mean! Yes good idea, I've added GHA caching with suitable conditions

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**